Yosemite/Yosemite/Stores/ReceiptStore.swift

```diff
@@ -217,6 +219,47 @@ private extension ReceiptStore {
             DDLogError("⛔️ Unable to save receipt for order id: \(order.orderID)")
         }
     }
+
+    /// Sends the receipt for the order to the provided email address.
+    /// Updates the billing address of the order to the provided email address and triggers the sending of the receipt.
+    /// - Parameters:
+    ///  - order: The order for which the receipt is being sent.
+    ///  - email: The email address to which the receipt is being sent.
+    ///  - onCompletion: The completion block to call when the operation is complete.
+    ///
+    func sendReceipt(order: Order, email: String, onCompletion: @escaping (Result<Order, Error>) -> Void) {
+        let updatedBillingAddress = order.billingAddress?.copy(email: email) ?? Address(firstName: "",
+                                                                                        lastName: "",
+                                                                                        company: nil,
+                                                                                        address1: "",
+                                                                                        address2: nil,
+                                                                                        city: "",
+                                                                                        state: "",
+                                                                                        postcode: "",
+                                                                                        country: "",
+                                                                                        phone: nil,
+                                                                                        email: email)
+        let orderToUpdate = order.copy(billingAddress: updatedBillingAddress)
+
+        let action = OrderAction.updateOrder(siteID: order.siteID, order: orderToUpdate, giftCard: nil, fields: [.billingAddress]) { result in
+            switch result {
+            case let .success(updatedOrder):
+                Task { [weak self] in
+                    guard let self else { return }
```

Comment:
nit: in case of self being nil, do we also want to invoke the completion block?
also a super nit on the indentation.